### DISCUSSION TITLE
Rearrange group order

### DISCUSSION THREAD

**Discussion by alexanderlindstrom** [home-assistant/frontend#18877] [Configuration]
I would like to be able to rearrange the order of a created group. For most cases it might not matter, but I am using an "entities" card which uses the exact order of the group, I'm sure there are more cases where this could be useful.
![reorder group](https://github.com/home-assistant/frontend/assets/1754686/03f95297-4b6a-461f-a190-8402823ef979)

**Comment by ildar170975** [comment]
It will need core changes since you need to define a kind of "index" of a member inside a group.
But what if you need to have some sorting on one card - and another sorting in another card?
So, this order should not be a property of a group.

**Reply by alexanderlindstrom** [reply]
If core lacks indexing for the entries in the list, that is an issue, but the list must give the entries a default index when created. So shouldn't the gui be able to generate new list with the indexes updated? Perhaps that is too much of a hack, so I understand that it might not be desirable. :)

To answer your question, I believe it's irrelevant, the card use case was just an example. Having multiple entities in a specific order have many other uses.

**Reply by karwosts** [reply]
I'm pretty sure it's already implicitly ordered, as the entity selector returns a list to the backend, which has an order.

You can already manually reorder the items by deleting the top one and re-adding it to the bottom of the list, and that ordering is preserved. So I guess there isn't any backend architectural change needed, would just need to add the grab handles to the entities in the frontend. I could maybe see that being a useful addition to the entity selector.
